core/primitives/src/version.rs

```diff
@@ -62,3 +63,88 @@ impl ProtocolVersionRange {
         self.lower <= version && self.upper.map_or(true, |upper| version < upper)
     }
 }
+
+/// New Protocol features should go here. Features are guarded by their corresponding feature flag.
+/// For example, if we have `ProtocolFeature::EVM` and a corresponding feature flag `evm`, it will look
+/// like
+/// ```
+/// #[cfg(feature = "evm")]
+/// EVM
+/// ```
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum ProtocolFeature {
+    #[cfg(feature = "protocol_feature_forward_chunk_parts")]
+    ForwardChunkParts,
+}
+
+/// Current latest stable version of the protocol.
+#[cfg(not(feature = "nightly_protocol"))]
+pub const PROTOCOL_VERSION: ProtocolVersion = 40;
+
+/// Current latest nightly version of the protocol.
+#[cfg(feature = "nightly_protocol")]
+pub const PROTOCOL_VERSION: ProtocolVersion = 41;
```

Comment:
There is a sanity check that the `PROTOCOL_VERSION` constant matches the protocol version in `genesis_config.json`. I assume it is there because a mismatch could lead to unexpected behaviour. Do we need to have a `nightly_genesis_config.json` and use it somehow when the binary is compiled with `nightly`?

Reply:
It is used mostly for testing, so yes we can do that when we update CI to include testing for nightly features.